crates/wasm-smith/src/lib.rs

```diff
@@ -1021,54 +1068,61 @@ where
+
+                    let mut global_idx = 0;
+                    for (_, _, imp) in &self.imports {
+                        match imp {
+                            Import::Global(g) => {
+                                if !g.mutable && g.val_type == ValType::I32 {
+                                    choices.push(Box::new(move |_| {
+                                        Ok(Instruction::GlobalGet(global_idx))
+                                    }));
+                                }
+                                global_idx += 1;
                             }
-                            global_idx += 1;
+                            _ => {}
                         }
-                        _ => {}
                     }
                 }
-            }
 
-            // Passive data can only be generated if bulk memory is enabled.
-            // Otherwise if there are no memories we *only* generate passive
-            // data. Finally if all conditions are met we use an input byte to
-            // determine if it should be passive or active.
-            let kind = if self.config.bulk_memory_enabled() && (memories == 0 || u.arbitrary()?) {
-                DataSegmentKind::Passive
-            } else {
-                let f = u.choose(&choices)?;
-                let offset = f(u)?;
-                let memory_index = u.int_in_range(0..=memories - 1)?;
-                DataSegmentKind::Active {
-                    offset,
-                    memory_index,
-                }
-            };
-            let init = u.arbitrary()?;
-            self.data.push(DataSegment { kind, init });
-            Ok(())
-        })
+                // Passive data can only be generated if bulk memory is enabled.
+                // Otherwise if there are no memories we *only* generate passive
+                // data. Finally if all conditions are met we use an input byte to
+                // determine if it should be passive or active.
+                let kind =
+                    if self.config.bulk_memory_enabled() && (memories == 0 || u.arbitrary()?) {
+                        DataSegmentKind::Passive
+                    } else {
+                        let f = u.choose(&choices)?;
+                        let offset = f(u)?;
+                        let memory_index = u.int_in_range(0..=memories - 1)?;
+                        DataSegmentKind::Active {
+                            offset,
+                            memory_index,
+                        }
+                    };
+                let init = u.arbitrary()?;
+                self.data.push(DataSegment { kind, init });
+                Ok(())
+            },
+        )
     }
 }
 
 pub(crate) fn arbitrary_loop(
     u: &mut Unstructured,
+    min: usize,
     max: usize,
     mut f: impl FnMut(&mut Unstructured) -> Result<()>,
 ) -> Result<()> {
-    for _ in 0..max {
+    for _ in min..max {
```

Comment:
This may not work as intended? I think perhaps for `0..min` this would unconditionally call `f` and then from `0..(max - min)` would conditionally call it?

(or `keep_going = i < min || ...`)

Reply:
Ah yes, also what @alexcrichton said :)

Reply:
D'oh; this is what I get for leaving my brain at home today. Thanks!